R/updateNullModCond.R

```diff
@@ -1,37 +1,40 @@
 
-### fit a conditional model based on a previously fit model. Default does not re-estimate variance components, but it may. 
+### fit a conditional model based on a previously fit model. Default does not re-estimate variance components, but it may.
 updateNullModCond <- function(nullmod, G, covMatList = NULL,  AIREML.tol = 1e-6, max.iter = 100, drop.zeros = TRUE, verbose = TRUE){
-    
-    ## a few checks that may be transfered to wrapper function: 
+
+    ## a few checks that may be transfered to wrapper function:
     if (nullmod$hetResid & is.null(nullmod$group.idx)) stop("group indices are required for updating the null model")
 
     #if (updateVarComp){ ## this check may be pulled out for wrapper function.
     if (is.null(covMatList)) stop("covMatList is needed for udpating variance components")
     if (nullmod$hetResid & is.null(nullmod$group.idx)) stop("group indices are required for updating variance components")
     #}
 
-    
+    if (is.null(colnames(G))) colnames(G) <- paste0("var_", 1:ncol(G))
```

Comment:
To add conditional variants to the model string, we need to make sure the genotype matrix has column names for the variant. If `G` has colnames, it will use those. If it does not, I used the convention from analysis_pipeline where they start with "var_", but because the variant.id is unknown, they are just numbered sequentially (e.g., var_1, var_2, var_3, etc.).